### PR TITLE
Don’t try to build compiled metrics, when loading the model

### PR DESCRIPTION
**Pull request by FLming**
Currently, the user can pass the strings to the metrics argument in `Model.compile` function, and convert it to one of `tf.keras.metrics` classes based on y_true shape and y_pred shape(In function `MetricsContainer._get_metric_object`). 

This feature brings us convenience, but when the user saves the model and loads it, the model is compiled by default, and at the same time, it tries to build the compiled metrics. But when the compiled metrics are built, only the output shape of the model is used to guess which metric class the user wants to use, which brings surprises. The current examples are sparse categorical accuracy and sparse categorical cross-entropy. 

Here is the small demo to reproduce the problem. I have encountered this problem. I always thought that the performance of the model was not good, but I did not expect the metric class to be wrong.

```python
# TensorFlow and tf.keras
import tensorflow as tf

# Helper libraries
import numpy as np

print(tf.__version__)

fashion_mnist = tf.keras.datasets.fashion_mnist

(train_images, train_labels), (test_images, test_labels) = fashion_mnist.load_data()

model = tf.keras.Sequential([
    tf.keras.layers.Flatten(input_shape=(28, 28)),
    tf.keras.layers.Dense(128, activation='relu'),
    tf.keras.layers.Dense(10)
])

model.compile(optimizer='adam',
              loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
              metrics=['accuracy'])

model.fit(train_images, train_labels, epochs=1)

test_loss, test_acc = model.evaluate(test_images,  test_labels, verbose=2)

model.save('model.h5')

model = tf.keras.models.load_model('model.h5')

test_loss, test_acc = model.evaluate(test_images,  test_labels, verbose=2)

model = tf.keras.models.load_model('model.h5', compile=False)

model.compile(optimizer='adam',
              loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
              metrics=['accuracy'])

test_loss, test_acc = model.evaluate(test_images,  test_labels, verbose=2)
```
```
1875/1875 [==============================] - 2s 1ms/step - loss: 3.1409 - accuracy: 0.6773
313/313 - 0s - loss: 0.8313 - accuracy: 0.7169
313/313 - 0s - loss: 0.8313 - accuracy: 0.1098
313/313 - 0s - loss: 0.8313 - accuracy: 0.7169
```

I also found that some users have encountered this "surprise" #43230

### CODE REVIEW

**Comment by FLming** [comment]
@googlebot I signed it!

**Comment by FLming** [comment]
The most direct way is to comment `model.compile metrics.build(model.outputs, model.outputs)`. I didn’t go deep into why it needs to try to build at this time, so I just deleted them.

**Comment by k-w-w** [comment]
Thanks for the PR, although this bug has been fixed in the latest release of TensorFlow: 5adacc88077ef82f6c4a7f9bb65f9ed89f9d8947
